### PR TITLE
Fix NRE in Test-Json

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Add null check.

## PR Context

The issue was discovered while nullable annotating in #11397 but I believe the _bug_ should be fixed in separate commit.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public class TestJsonCommand : PSCmdlet
         /// <returns>Return value is unreachable since we always rethrow.</returns>
         private static bool UnwrapException(Exception e)
         {
-            if (e is TargetInvocationException)
+            if (e is TargetInvocationException && e.InnerException != null)
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs:62]: I think this is just easier to understand.
```suggestion
            if (e.InnerException != null && e is TargetInvocationException)
```

**Review by TravisEz13 — DISMISSED** [review]
consider suggestion and rebase

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs:62]: Done.

**Comment by TravisEz13** [comment]
@PoshChan Please remind me 1 day

**Comment by PoshChan** [comment]
@TravisEz13, I do not understand: Please remind me 1 day

<details>
<summary>Commands available in this repo for you:
</summary>
<ul><li>
<code>retry &lt;target&gt;</code> this will attempt to retry only the failed jobs for the target pipeline, <code>restart</code> can be used in place of <code>retry</code>
</li>
<li>
<code>rebuild &lt;target&gt;</code> this will perform a complete rebuild of the target pipeline, <code>rerun</code> can be used in place of <code>rebuild</code> Supported values for &lt;target&gt; which can be a comma separated list are: <code>static</code>,<code>windows</code>,<code>macos</code>,<code>linux</code>,<code>ssh</code>,<code>all</code>
</li>
<li>
<code>get failures</code> this will attempt to get the latest failures for all of the target pipelines
</li>
<li>
<code>remind me in &lt;value&gt; &lt;units&gt;</code> this will create a reminder that will be posted after the specified duration &lt;value&gt; is a number, and &lt;units&gt; can be <code>minutes</code>, <code>hours</code>, or <code>days</code> (singular or plural)
</li>

</ul>
</details>

**Comment by iSazonov** [comment]
@TravisEz13 Can you merge the PR?

**Comment by iSazonov** [comment]
@TravisEz13 Friendly ping.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ public class TestJsonCommand : PSCmdlet
         /// <returns>Return value is unreachable since we always rethrow.</returns>
         private static bool UnwrapException(Exception e)
         {
-            if (e is TargetInvocationException)
+            if (e.InnerException != null && e is TargetInvocationException)
             {
                 ExceptionDispatchInfo.Capture(e.InnerException).Throw();
             }
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs:65]: Do we need to add an else?
cc @daxian-dbw

Comment (daxian-dbw) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs:65]: The check `e.InnerException != null` is not needed. The inner exception of `TargetInvocationException` is always not null.

See https://docs.microsoft.com/en-us/dotnet/api/system.reflection.targetinvocationexception?view=netcore-3.1
> When created, the TargetInvocationException is passed a reference to the exception thrown by the method invoked through reflection. The InnerException property holds the underlying exception.

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs:65]: It is annotated as nullable and compiler warns us
https://source.dot.net/#System.Private.CoreLib/TargetInvocationException.cs,12

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**